hybig/browse.py

```diff
@@ -233,26 +237,49 @@ def convert_mulitband_to_raster(data_array: DataArray) -> ndarray:
 
     bands = data_array.to_numpy()
 
-    # Create an alpha layer where input NaN values are transparent.
+    if data_array.rio.count == 4:
+        return convert_to_uint8(bands, original_dtype(data_array))
+
+    # Input NaNs in any of the RGB bands are made transparent.
     nan_mask = np.isnan(bands).any(axis=0)
     nan_alpha = np.where(nan_mask, TRANSPARENT, OPAQUE)
 
-    # grab any existing alpha layer
-    bands, image_alpha = remove_alpha(bands)
+    raster = convert_to_uint8(bands, original_dtype(data_array))
 
-    norm = Normalize(vmin=np.nanmin(bands), vmax=np.nanmax(bands))
-    raster = np.nan_to_num(np.around(norm(bands) * 255.0), copy=False, nan=0.0).astype(
-        'uint8'
-    )
+    return np.concatenate((raster, nan_alpha[None, ...]), axis=0)
+
+
+def convert_to_uint8(bands: ndarray, dtype: str | None) -> ndarray[uint8]:
+    """Convert Banded data with NaNs (missing) into a uint8 data cube.
+
+    99.99% of the time this will simply pass through the data coercing it back
+    into unsigned ints and setting the missing values to 0 that will be masked
+    as transparent in the output png.
+
+    There is a some small non-zero chance that the input RGB image was 16-bit
+    and if any of the values exceed 255, we must normalize all of input data to
+    the range 0-255.
+
+    """
 
-    if image_alpha is not None:
-        # merge missing alpha with the image alpha band prefering transparency
-        # to opaqueness.
-        alpha = np.minimum(nan_alpha, image_alpha).astype(np.uint8)
+    if dtype != 'uint8' and np.nanmax(bands) > 255:
+        norm = Normalize(vmin=np.nanmin(bands), vmax=np.nanmax(bands))
+        scaled = np.around(norm(bands) * 255.0)
+        raster = scaled.filled(0).astype('uint8')
     else:
-        alpha = nan_alpha
+        raster = np.nan_to_num(bands).astype('uint8')
 
-    return np.concatenate((raster, alpha[None, ...]), axis=0)
+    return raster
+
+
+def original_dtype(data_array: DataArray) -> str | None:
+    """Return the original input data's type.
+
+    rastero_optn retains the input dtype in the encoding dictionary and is used
```

Comment:
It's sort of used directly, because data_array is the return value of `rasterio_open()`
This is just a convenience function to know what the datatype was on the original input file.